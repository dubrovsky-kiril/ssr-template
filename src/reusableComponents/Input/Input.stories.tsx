import * as React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import Input from "./Input";

storiesOf("Input", module).add(
  "default",
  withInfo()(() => <Input id="productId" label="Product id" type="text" />)
);
