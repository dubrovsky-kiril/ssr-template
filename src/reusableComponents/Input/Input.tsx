import * as React from "react";

type InputTypes = {
  /** connects label with input */
  id: string;
  /** text for inputs label */
  label: string;
  /** type of input (text, number etc.) */
  type: string;
  /** forbids empty input */
  isRequired?: boolean;
};

const Input: React.FunctionComponent<InputTypes> = ({
  id,
  label,
  type,
  isRequired = false
}) => {
  return (
    <div>
      <label htmlFor={id}>{label}</label>
      <input id={id} name={id} type={type} required={isRequired} />
    </div>
  );
};

export default Input;
