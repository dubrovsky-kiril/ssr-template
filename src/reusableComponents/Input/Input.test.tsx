import * as React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Input from "./Input";

describe("Input", () => {
  it("should apply 'id' prop to labels htmlFor attr and inputs id attr", () => {
    const wrapper = shallow(
      <Input id="productId" label="Product id" type="text" />
    );

    expect(wrapper.find("label").props().htmlFor).toBe("productId");
    expect(wrapper.find("input").props().id).toBe("productId");
  });
  it("should apply 'label' prop to labels child", () => {
    const wrapper = shallow(
      <Input id="productId" label="Product id" type="text" />
    );

    expect(wrapper.find("label").text()).toEqual("Product id");
  });
  it("should apply 'type' prop to input", () => {
    const wrapper = shallow(
      <Input id="productId" label="Product id" type="text" />
    );

    expect(wrapper.find("input").props().type).toBe("text");
  });
  it("should apply 'isRequired' prop to input", () => {
    const wrapper = shallow(
      <Input id="productId" label="Product id" type="text" />
    );

    expect(wrapper.find("input").props().required).toBe(false);

    wrapper.setProps({ isRequired: true });

    expect(wrapper.find("input").props().required).toBe(true);
  });
  it("should match snapshot", () => {
    const tree = shallow(
      <Input id="productId" label="Product id" type="text" />
    );

    expect(toJson(tree)).toMatchSnapshot();
  });
});
