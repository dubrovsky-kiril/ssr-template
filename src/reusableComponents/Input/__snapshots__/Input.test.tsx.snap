// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<div>
  <label
    htmlFor="productId"
  >
    Product id
  </label>
  <input
    id="productId"
    name="productId"
    required={false}
    type="text"
  />
</div>
`;
